### ISSUE TITLE
Document SpringSessionRedisConnectionFactory

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-session#2102]
While looking at [this Spring Boot issue](https://github.com/spring-projects/spring-boot/issues/31510), I learned of the existence of `@SpringSessionRedisConnectionFactory` by noticing its usage in `RedisHttpSessionConfiguration`. It would be good if the annotation was mentioned in the reference documentation.

**Comment by AndreynRosa** [comment]
Hello @wilkinsona and @marcusdacoregio,  can I take on this task to give it a try?

**Comment by marcusdacoregio** [comment]
Yes @AndreynRosa, the ticket is yours. I think that a new section could be added into https://docs.spring.io/spring-session/reference/configuration/redis.html approaching the use case where a custom `@ SpringSessionRedisConnectionFactory` is needed.

**Comment by marcusdacoregio** [comment]
Hi @AndreynRosa, have you been able to work on this issue?
